switch from bitly to shlink

theofficialgman opened the pull request:
keep bitly_link function and point it to shlink_link incase an old instance of the manage script is still running with the new api sourced

I would like this to stay as a draft until I can see shlink results in the pi-apps-analytics repo for all apps. that will take 1-2 days

(edit: force pushed to add relevant co-authors for this change)

Botspot commented:
The `etc/setting-params/Enable analytics` file needs its caption updated.

theofficialgman commented:
for any users that happen to come across this PR and are may be concerned about security implications of this change, we still do NOT collect any personally identifying information as part of analytics. 
Below is sample visit output from one of the short urls on pi-apps shlink server:
```
{
  "visits": {
    "data": [
      {
        "referer": "",
        "date": "2022-08-03T20:06:00+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T20:05:09+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T15:39:24+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T15:39:23+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T14:44:15+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T14:44:14+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T14:44:13+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T14:41:50+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      },
      {
        "referer": "",
        "date": "2022-08-03T14:41:48+00:00",
        "userAgent": "Pi-Apps Raspberry Pi app store",
        "visitLocation": null,
        "potentialBot": false
      }
    ],
    "pagination": {
      "currentPage": 1,
      "pagesCount": 1,
      "itemsPerPage": 9,
      "itemsInCurrentPage": 9,
      "totalItems": 9
    }
  }
}
```

Botspot commented:
I've reworked the documentation for the shlink_link function. In an attempt to improve grammatical flow, I removed the credit for who wrote one of the github actions scripts. As I uncredited both myself and you, @theofficialgman, nothing seems unfair, but if you feel otherwise, feel free to add it back.

theofficialgman commented:
> I've reworked the documentation for the shlink_link function. In an attempt to improve grammatical flow, I removed the credit for who wrote one of the github actions scripts. As I uncredited both myself and you, @theofficialgman, nothing seems unfair, but if you feel otherwise, feel free to add it back.

sounds good, I don't care either way but also felt like it needed to change. the commit history is enough credit for users that care to know

Botspot commented:
> sounds good, I don't care either way but also felt like it needed to change.

Correct, leaving it as "a script that botspot wrote..." would no longer be accurate. Honestly, I'm looking forward to retire my xdotool-abusing monstrosity of a script!

theofficialgman commented:
all links are functioning at the analytics repo (the debug output in the log would be `null` if it wasn't working) https://github.com/Botspot/pi-apps-analytics/runs/7664000849?check_suite_focus=true

this can be merged
